packages/compiler/src/init/init-template.ts

```diff
@@ -15,6 +15,13 @@ export interface InitTemplateInput {
 }
 
 export interface InitTemplate {
+  /**
+   * The kind of project this tempolate initialize. This will change things like where dependencies are added.
+   * For example, a library will add dependencies to `peer` and `dev` dependencies, while a project will add them to `dependencies`.
+   * @default "project"
+   */
+  target?: "library" | "project";
```

Comment:
open to different name or a different approach (more specialized), thought of 
- `kind`
- `preset`
- `usePeerDependencies` for a more specialized option but I would prefer to keep it more generic instead of adding a million tiny knobs